### PR TITLE
Convert vaccination status store to VUE 3.0 for AB#15709

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15709](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15709)

## Description

- Created vaccinationStatusAuthenticated store from old vaccinationStatus store
- Created vaccinationStatusPublic store from old vaccinationStatus store
- Kept getter/computed and action names the same
- Updated log message in Note store to match all other stores

**Note:** Reason for naming vaccinationStatusAuthenticated and vaccinationStatusPublic was to keep the files next to one another - If this is not wanted, suggest a new name for those files.

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        const vaccinationStatusService =
+            container.get<IVaccinationStatusService>(
+                SERVICE_IDENTIFIER.VaccinationStatusService
+            );
+
+        const errorStore = useErrorStore();
+
+        interface PublicVaccinationStatus {
+            vaccinationStatus?: VaccinationStatus;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts:27]: Suggest renaming `vaccinationStatus` property to `data` to match other stores.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            error?: CustomBannerError;
+            status: LoadStatus;
+            statusMessage: string;
+        }
+
+        interface PublicVaccinationRecord {
+            vaccinationRecord?: CovidVaccineRecord;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts:34]: Suggest renaming `vaccinationRecord` property to `data` to match other stores.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts:1]: I think `public` can be dropped from the names of everything inside the store (interfaces, refs, computed refs, functions) now that it's been split up.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:1]: I think `authenticated` can be dropped from the names of everything inside the store (interfaces, refs, computed refs, functions) now that it's been split up.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        const defaultVaccineRecordState: VaccineRecordState = {
+            hdid: "",
+            download: false,
+            status: LoadStatus.NONE,
+            statusMessage: "",
+            resultMessage: "",
+        };
+
+        interface AuthenticatedVaccinationStatus {
+            vaccinationStatus?: VaccinationStatus;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:37]: Suggest renaming `vaccinationStatus` property to `data` to match other stores.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+import { ResultError } from "@/models/errors";
+import { LoadStatus } from "@/models/storeOperations";
+import VaccinationStatus from "@/models/vaccinationStatus";
+import VaccineRecordState from "@/models/vaccineRecordState";
+import { ILogger, IVaccinationStatusService } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useVaccinationStatusStore = defineStore(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:17]:
```suggestion
export const useVaccinationStatusAuthenticatedStore = defineStore(
```

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+import { DateWrapper, StringISODate } from "@/models/dateWrapper";
+import { CustomBannerError, ResultError } from "@/models/errors";
+import { LoadStatus } from "@/models/storeOperations";
+import VaccinationStatus from "@/models/vaccinationStatus";
+import { ILogger, IVaccinationStatusService } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useVaccinationStatusStore = defineStore(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts:15]:
```suggestion
export const useVaccinationStatusPublicStore = defineStore(
```

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+                        ErrorType.Retrieve
+                    );
+                });
+        }
+
+        function authenticatedVaccineRecordState(
+            hdid: string
+        ): VaccineRecordState {
+            return getAuthenticatedVaccineRecordState(hdid);
+        }
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:370]: I think this should be grouped with the getters.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+        function retrieveAuthenticatedVaccineRecord(
+            hdid: string
+        ): Promise<void> {
+            const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+            const vaccinationStatusService =
+                container.get<IVaccinationStatusService>(
+                    SERVICE_IDENTIFIER.VaccinationStatusService
+                );
+
+            logger.debug(`Retrieving authenticated vaccination record`);
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:315]:
```suggestion
            logger.debug(`Retrieving authenticated vaccination record`);
```

These services have already been retrieved.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts (hunk 1 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts]: Changed.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts]: Changed.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts (hunk 3 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts]: Grouped under computed.

Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts]: Changed.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts (hunk 6 above)
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts]: Grouped under computed.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts:1]: Changed.

File: Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts
Comment (BrianMaki) [reply Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts:1]: Changed.

**Review by MikeLyttle — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**